testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,607 @@
+        with DCDFile(self.dcdfile) as dcd:
+            dcd.seek(self.new_frame)
+            assert_equal(dcd.tell(), self.new_frame)
+
+    @raises(IOError)
+    def test_seek_negative(self):
+        # frame seek with negative number
+        with DCDFile(self.dcdfile) as dcd:
+            dcd.seek(-78)
+
+    def test_iteration(self):
+        with DCDFile(self.dcdfile) as dcd:
+            for _ in range(self.num_iters):
+                dcd.__next__()
+            assert_equal(dcd.tell(), self.num_iters)
+
+    def test_zero_based_frames(self):
+        expected_frame = 0
+        with DCDFile(self.dcdfile) as dcd:
+            assert_equal(dcd.tell(), expected_frame)
+
+    def test_length_traj(self):
+        expected = self.traj_length
+        with DCDFile(self.dcdfile) as dcd:
+            assert_equal(len(dcd), expected)
+
+    @raises(IOError)
+    def test_open_wrong_mode(self):
+        DCDFile('foo', 'e')
+
+    @raises(IOError)
+    def test_raise_not_existing(self):
+        DCDFile('foo')
+
+    def test_natoms(self):
+        with DCDFile(self.dcdfile) as dcd:
+            assert_equal(dcd.header['natoms'], self.natoms)
+
+    @raises(IOError)
+    @run_in_tempdir()
+    def test_read_write_mode_file(self):
+        with DCDFile('foo', 'w') as f:
+            f.read()
+
+    @raises(IOError)
+    def test_read_closed(self):
+        with DCDFile(self.dcdfile) as dcd:
+            dcd.close()
+            dcd.read()
+
+    def test_iteration_2(self):
+        with DCDFile(self.dcdfile) as dcd:
+            with dcd as f:
+                for i, _ in enumerate(f):
+                    assert_equal(i + 1, f.tell())
+                # second iteration should work from start again
+                for i, _ in enumerate(f):
+                    assert_equal(i + 1, f.tell())
+
+
+class TestDCDWriteHeader(object):
+    def setUp(self):
+        self.tmpdir = tempdir.TempDir()
+        self.testfile = self.tmpdir.name + '/test.dcd'
+        self.dcdfile = DCD
+
+    def tearDown(self):
+        try:
+            os.unlink(self.testfile)
+        except OSError:
+            pass
+        del self.tmpdir
+
+    def test_write_header(self):
+        # test that _write_header() can produce a very crude
+        # header for a new / empty file
+        with DCDFile(self.testfile, 'w') as dcd:
+            dcd.write_header(
+                remarks='Crazy!',
+                natoms=22,
+                istart=12,
+                nsavc=10,
+                delta=0.02,
+                charmm=1)
+
+        # we're not actually asserting anything, yet
+        # run with: nosetests test_libdcd.py --nocapture
+        # to see printed output from nose
+        with DCDFile(self.testfile) as dcd:
+            header = dcd.header
+            assert_equal(header['remarks'], 'Crazy!')
+            assert_equal(header['natoms'], 22)
+            assert_equal(header['istart'], 12)
+            assert_equal(header['charmm'], 5)
```

Comment:
This assert for `charmm` should *fail*, shouldn't it? Otherwise it means that  `DCD_IS_CHARMM` 0x01 and `HAS_EXTRA_BLOCKS` 0x04 are both set, even though you haven't set `HAS_EXTRA_BLOCKS` explicitly.